File: [2] Calculations/Paint-Calculator/PaintCalculator.py
# Calculate gallons of paint needed to paint the ceiling of a room.
# Prompt for the length and width, and assume one gallon covers 350 square feet.
# Display the number of gallons needed to paint the ceiling as a whole number.
import math


# Revise the program to ensure that inputs are entered as numeric values.
# Don’t allow the user to proceed if the value entered is not numeric.
def ValidatedNumber(prompt, _min=None, _max=None):
    while True:
        user_input = input(prompt)
        try:
            number = float(user_input)
            if (_min is None or _min <= number) and (_max is None or _max >= number):
                return number
            else:
                print(f"Input out of range. [{_min}-{_max}]")
        except ValueError:
            print("! Error: Input is NaN")


# Use a constant to hold the conversion rate.
ONE_GALLON = 350  # square feet


class PaintGallons:
    def __init__(self, room_type=None):
        if room_type == "square":
            self.length = ValidatedNumber("Length: ")
            self.width = ValidatedNumber("Width: ")
            self.area = self.length * self.width
        # Implement support for a round room.
        elif room_type == "round":
            self.radius = ValidatedNumber("Radius: ")
            self.area = math.pi * (self.radius ** 2)
        # Implement support for an L-shaped room.
        elif room_type == "l-shaped":
            self.length1 = ValidatedNumber("Length of the first rectangle: ")
            self.width1 = ValidatedNumber("Width of the first rectangle: ")
            self.area1 = self.length1 * self.width1

            self.length2 = ValidatedNumber("Length of the second rectangle: ")
            self.width2 = ValidatedNumber("Width of the second rectangle: ")
            self.area2 = self.length2 * self.width2

            self.area = self.area1 + self.area2

        # Ensure that you round up to the next whole number.
        self.gallons = math.ceil(self.area / ONE_GALLON)

        if self.gallons <= 1:
            print(f"You will need to purchase 1 gallon of paint ")
        else:
            print(f"You will need to purchase {self.gallons} gallons of paint ")
        print(f"to cover {self.area} square feet.")


print("[1] Square/Rectangular")
print("[2] Round")
print("[3] L-Shaped")
choice = ValidatedNumber("Enter your room type: ", 1, 3)
if choice == 1:
    paint_buckets = PaintGallons("square")
elif choice == 2:
    paint_buckets = PaintGallons("round")
elif choice == 3:
    paint_buckets = PaintGallons("l-shaped")

File: VB.Net/Calculations/Paint-Calculator/PaintCalculator.vb
Public Class PaintCalculator
    Private _area = 1, _gallons = 1, _plurality = ""
    Private Const GALLONS = 350 'Square Feet

    Private Sub UpdateOnChange(sender As Object, e As EventArgs) Handles RadioButton1.CheckedChanged, RadioButton2.CheckedChanged, RadioButton3.CheckedChanged, NumericUpDown1.ValueChanged, NumericUpDown2.ValueChanged, NumericUpDown3.ValueChanged, NumericUpDown4.ValueChanged
        If NumericUpDown1.Value <> 0 And NumericUpDown2.Value <> 0 And NumericUpDown3.Value <> 0 And NumericUpDown4.Value <> 0 Then
            If RadioButton1.Checked Then
                _area = Rectangular()
            ElseIf RadioButton2.Checked Then
                _area = Round()
            Else
                _area = LShaped()
            End If

            _gallons = Math.Ceiling(_area / GALLONS)
            _plurality = If(_gallons <= 1, "gallon", "gallons")
        End If

        TextBox1.Text = $"You will need to purchase {_gallons} {_plurality} of paint to cover {_area:F0} square feet."
    End Sub

    Private Function Rectangular()
        Label1.Text = "Length"
        Label2.Text = "Width"
        Label2.Visible = True
        NumericUpDown2.Visible = True
        Label3.Visible = False
        NumericUpDown3.Visible = False
        Label4.Visible = False
        NumericUpDown4.Visible = False

        Dim _length As Integer = NumericUpDown1.Value
        Dim _width As Integer = NumericUpDown2.Value
        Return _length * _width
    End Function

    Private Function Round()
        Label1.Text = "Radius"
        Label2.Visible = False
        NumericUpDown2.Visible = False
        Label3.Visible = False
        NumericUpDown3.Visible = False
        Label4.Visible = False
        NumericUpDown4.Visible = False

        Dim _radius As Integer = NumericUpDown1.Value
        Return Math.PI * _radius
    End Function

    Private Function LShaped()
        Label1.Text = "Length 1"
        Label2.Text = "Width 1"
        Label2.Visible = True
        NumericUpDown2.Visible = True
        Label3.Visible = True
        NumericUpDown3.Visible = True
        Label4.Visible = True
        NumericUpDown4.Visible = True

        Dim _length1 As Integer = NumericUpDown1.Value
        Dim _width1 As Integer = NumericUpDown2.Value
        Dim _length2 As Integer = NumericUpDown3.Value
        Dim _width2 As Integer = NumericUpDown4.Value
        Return (_length1 * _width1) + (_length2 * _width2)
    End Function
End Class